### DISCUSSION TITLE
1.0.1 is now LIVE!

### DISCUSSION THREAD

**Discussion by gabbybilka** [microsoft/WindowsAppSDK#2291] [Announcements]
## Windows App SDK 1.0.1 is now available

Windows App SDK 1.0.1 is a stable servicing release that includes critical bug fixes and multi-window support for the 1.0 release. 

- If you're building an app with the Windows App SDK for the first time, follow the [installation instructions here](https://docs.microsoft.com/en-us/windows/apps/winui/winui3/create-your-first-winui3-app?tabs=desktop-csharp).
- If you are already using the Windows App SDK 1.0 and would like to update to the latest release:

    - Update your **Microsoft.WindowsAppSDK** NuGet package reference from **1.0.0** to **1.0.1**.
    - [Download the 1.0.1 runtime](https://aka.ms/windowsappsdk/downloads) using either the standalone Runtime installer (**.exe**) or the Windows App Runtime Redistributable which includes all MSIX package dependencies.

### Bugs fixed in Windows App SDK 1.0.1:

- Fixed issue causing the MddBootstrapAutoinitializer to not compile with enabled ImplicitUsings: #1686
- Fixed issue where focus in WebView2 would be unexpectedly lost causing input and selection issues: [#5615](https://github.com/microsoft/microsoft-ui-xaml/issues/5615) & [#5570](https://github.com/microsoft/microsoft-ui-xaml/issues/5570)
- Fixed issue causing the in-app toolbar in Visual Studio to be unclickable when using a custom title bar in a WinUI 3 app.
- Fixed issue causing Snap Layout to not appear when using a custom title bar in a WinUI 3 app: [#6333](https://github.com/microsoft/microsoft-ui-xaml/issues/6333) & [#6246](https://github.com/microsoft/microsoft-ui-xaml/issues/6246)
- Fixed issue causing an exception when setting Window.ExtendsContentIntoTitleBar property when Window.SetTitlebar has been called with a still-loading UIElement.
- Fixed issue where Single-project MSIX apps did not support `dotnet build`.
- Fixed issue causing unpackaged apps to not install after installing a packaged app: #1871
- Fixed issue reducing performance during mouse drag operations.
- Fixed crash when calling GetWindowIdFromWindow() in unpackaged apps: Discussed in #1891

Additionally, for apps with custom title bars, we have made changes in this release (and fixed numerous issues) that include fixes to the glass window used for drag&drop operations. 
The recommendation is to use the default values and behaviors (give them a try!). 
If your title bar used margins so that the default caption buttons were interactive, we recommend visualizing your drag region by setting the background of your title bar to red and then adjusting the margins to extend the drag region to the caption controls.

### New capabilities
We have stabilized and enabled the creation of **multiple windows on the same thread** in WinUI 3 applications. Previously, this would produce an error.

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/WindowsAppSDK/releases/tag/v1.0.1'>1.0.1</a>.</em>

**Comment by DominicMaas** [comment]
The release notes for 0.8.6 & 0.8.7-preview1 mention performance improvements for C#/.NET Applications via C#/WinRT. Are these performance improvements / newer versions also bundled in 1.0.1? 

(I may have a fundamental misunderstanding with the relationship between WinUI, C#/WinRT, and Windows App SDK. Not immediately clear what includes what, how things are versioned etc.)

**Reply by ghost** [reply]
> (I may have a fundamental misunderstanding with the relationship between WinUI, C#/WinRT, and Windows App SDK.)

"_Windows App SDK is a set of libraries, frameworks, components, and tools that you can use in your apps to access powerful Windows platform functionality from all kinds of apps on many versions of Windows_."

WinUI **3** = is an UI framework , a component of Windows App SDK umbrella.

C#/WinRT = C# binding to the WinRT/COM ABI. 

> Are these performance improvements / newer versions also bundled in 1.0.1?

No. Language projections like C#/WinRT and C++/WinRT  are provided through Nuget. you have to reference those nuget packages in your solution.

**Reply by ghost** [reply]
**Relation between C#/WinRT and Windows App SDK : --**

(any) WinRT APIs = (any) **APIs** (written in any language) **exposed through the WinRT/COM ABI.**
Now, Windows App SDK's APIs are WinRT APIs, not C based Win32 APIs.
if anyone wants to call these _WinAppSDK's    WinRT APIs_ through the C# language , they will have to use the C#/WinRT   projection/binding/mapping. 

the same applies for C++/WinRT or Windows-rs and their relation with Windows App SDK.

**Reply by angelazhangmsft** [reply]
> The release notes for 0.8.6 & 0.8.7-preview1 mention performance improvements for C#/.NET Applications via C#/WinRT. Are these performance improvements / newer versions also bundled in 1.0.1?

The C# performance improvements mentioned in 0.8.6 and 0.8.7-preview1 will be available in an upcoming servicing update to 1.0.x. 

> (I may have a fundamental misunderstanding with the relationship between WinUI, C#/WinRT, and Windows App SDK. Not immediately clear what includes what, how things are versioned etc.)

C#/WinRT is used by components in the Windows App SDK components to generate the C# projections/APIs. The C#/WinRT NuGet isn't required in your application, since the projections are already pre-generated.

**Comment by ghost** [comment]
@gabbybilka , 
> * Fixed issue where Single-project MSIX apps did not support `dotnet build.`

 and https://portal.productboard.com/winappsdk/1-windows-app-sdk/c/57-support-dotnet-build 

are NOT same, yes ?

**Comment by shelllet** [comment]
upgrade to 1.0.1 made my app not work.
![image](https://user-images.githubusercontent.com/16729017/158985528-cb1583ad-0155-4704-9adf-963e54c32fdc.png)

**Reply by shelllet** [reply]
D:\a\_work\1\s\dev\WindowsAppRuntime_BootstrapDLL\MddBootstrap.cpp(536)\Microsoft.WindowsAppRuntime.Bootstrap.dll!00007FFA77E438F9: (caller: 00007FFA77E4244F) LogHr(1) tid(4324) 80040010 Object is not in any of the inplace active states
    Msg:[Bootstrap.Intitialize: Scanning packages for Major.Minor=1.0, Tag=, MinVersion=1.440.209.0] 
'App1.exe' (Win32): Loaded 'C:\Windows\System32\Windows.StateRepositoryCore.dll'. 
D:\a\_work\1\s\dev\WindowsAppRuntime_BootstrapDLL\MddBootstrap.cpp(625)\Microsoft.WindowsAppRuntime.Bootstrap.dll!00007FFA77E4407C: (caller: 00007FFA77E4244F) LogHr(2) tid(4324) 80040011 Not able to convert object
    Msg:[Bootstrap.Intitialize: Microsoft.WinAppRuntime.DDLM.0.319.455.0-x6_0.319.455.0_x64__8wekyb3d8bbwe not applicable. Version doesn't match MinVersion criteria (Major.Minor=1.0, Tag=, MinVersion=1.440.209.0)] 
D:\a\_work\1\s\dev\WindowsAppRuntime_BootstrapDLL\MddBootstrap.cpp(665)\Microsoft.WindowsAppRuntime.Bootstrap.dll!00007FFA77E44C4C: (caller: 00007FFA77E4244F) Exception(1) tid(4324) 80070491 There was no match for the specified key in the index.
    Msg:[Enumeration: Major.Minor=1.0, Tag=, MinVersion=1.440.209.0] 
Exception thrown at 0x00007FFADCD9478C in App1.exe: Microsoft C++ exception: wil::ResultException at memory location 0x00000062ED0FC3E0.
Exception thrown at 0x00007FFADCD9478C in App1.exe: Microsoft C++ exception: [rethrow] at memory location 0x0000000000000000.
D:\a\_work\1\s\dev\WindowsAppRuntime_BootstrapDLL\MddBootstrap.cpp(125)\Microsoft.WindowsAppRuntime.Bootstrap.dll!00007FFA77E53E61: (caller: 00007FF7713385BB) ReturnHr(1) tid(4324) 80070491 There was no match for the specified key in the index.
    Msg:[D:\a\_work\1\s\dev\WindowsAppRuntime_BootstrapDLL\MddBootstrap.cpp(665)\Microsoft.WindowsAppRuntime.Bootstrap.dll!00007FFA77E44C4C: (caller: 00007FFA77E4244F) Exception(1) tid(4324) 80070491 There was no match for the specified key in the index.
    Msg:[Enumeration: Major.Minor=1.0, Tag=, MinVersion=1.440.209.0] 
]

**Reply by castorix** [reply]
> D:\a_work\1\s\dev\WindowsAppRuntime_BootstrapDLL\MddBootstrap.cpp(...

You must install **WindowsAppRuntimeInstall.exe** for the new version...

**Reply by shelllet** [reply]
I can not find the newer WindowsAppRuntimeInstall.exe.

Where can i found the 1.0.1 version of WindowsAppRuntimeInstall.exe ?

**Reply by ghost** [reply]
Microsoft's docs are pure Dog sh!te. 

Use the horizontal scrollbar to find the Installer and MSIX(s) files to download.
https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/downloads#current-releases

**Comment by ghost** [comment]
@**whoever is WindowsAppSDK-PM now**,  Please mention the following link in your release notes. 

https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/downloads#current-releases
[Offcourse Use the horizontal scrollbar to find the Installer.exe and MSIX(s) files to download.]

**Reply by nkasco** [reply]
Agreed. In hindsight this makes complete sense to need to do, but it was not obvious up front. Running Update-Package broke my app too until I installed the new runtime.

**Reply by angelazhangmsft** [reply]
Thanks for the feedback. The release notes have been updated to include the link to the downloads page for the required runtime downloads. We'll make sure to link to this for future releases as well!

**Comment by Samuel12321** [comment]
Does this release include a fix for `ShouldConstrainToRootBounds="False"` for WinUI 3?

**Reply by castorix** [reply]
> Does this release include a fix for `ShouldConstrainToRootBounds="False"` for WinUI 3?

I tested on Windows 10 21H1 with a Popup Context Menu and it still does not work...
